### PR TITLE
gltf model upload UI WIP

### PR DESCRIPTION
**Pull request by akleshchev**
This is a prototype limited to some regions for some preview and considerations with access from develop menu, not a finished implementation.
image

### CODE REVIEW

**Review by marchcat — APPROVED** [review]
The year in the licenses is a bit inconsistent. Otherwise, looks good.

**Comment by akleshchev** [comment]
Thank you, will fix once I get 'all green' from builds.

**Comment by akleshchev** [comment]
Done.

File: indra/newview/llfloatergltfasseteditor.cpp [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+    LLScrollContainer::Params scroller_params(LLUICtrlFactory::getDefaultParams<LLFolderViewScrollContainer>());
+    scroller_params.rect(scroller_view_rect);
+    scroller_params.name("folder_scroller");
+    mScroller = LLUICtrlFactory::create<LLFolderViewScrollContainer>(scroller_params);
+    mScroller->setFollowsAll();
+
+    // Insert that scroller into the panel widgets hierarchy
+    mItemListPanel->addChild(mScroller);
+
+    // Create the root model and view for all conversation sessions
```
Comment (maxim-productengine) [inline indra/newview/llfloatergltfasseteditor.cpp:123]: A little bit misleading comment

Comment (akleshchev) [reply indra/newview/llfloatergltfasseteditor.cpp:123]: Yeah, will fix, I used conversation floater as a base.

File: indra/newview/llfloatergltfasseteditor.cpp [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+        setTransformsEnabled(false);
+        return;
+    }
+
+    LLFolderViewItem* item = items.front();
+    LLGLTFFolderItem* vmi = static_cast<LLGLTFFolderItem*>(item->getViewModelItem());
+
+    switch (vmi->getType())
+    {
+    case LLGLTFFolderItem::TYPE_NODE:
```
Comment (maxim-productengine) [inline indra/newview/llfloatergltfasseteditor.cpp:332]: I'm not sure I like using switch just for one case, but up to you.

**Review by maxim-productengine — APPROVED** [review]

**Pull request merged**